ggshield/cmd/iac/scan/diff.py

```diff
@@ -105,14 +110,16 @@ def scan_diff_cmd(
         staged = False
 
     result = iac_scan_diff(ctx, directory, ref, staged)
-    scan = IaCDiffScanCollection(id=str(directory), result=result)
     output_handler = create_output_handler(ctx)
+    if isinstance(result, IaCSkipDiffScanResult):
+        return output_handler.process_skip_diff_scan()
+    scan = IaCDiffScanCollection(id=str(directory), result=result)
     return output_handler.process_diff_scan(scan)
 
 
 def iac_scan_diff(
     ctx: click.Context, directory: Path, ref: str, include_staged: bool
-) -> Optional[IaCDiffScanResult]:
+) -> Optional[Union[IaCDiffScanResult, IaCSkipDiffScanResult]]:
```

Comment:
nit: this can be written as:
```suggestion
) -> Union[IaCDiffScanResult, IaCSkipDiffScanResult, None]:
```